utils/kafka-utils/src/main/scala/pl/touk/nussknacker/engine/kafka/KafkaConfig.scala

```diff
@@ -25,10 +24,16 @@ case class KafkaConfig(kafkaAddress: String,
                        // and all topics related to this config require both key and value schema definitions.
                        useStringForKey: Boolean = true,
                        schemaRegistryCacheConfig: SchemaRegistryCacheConfig = SchemaRegistryCacheConfig(),
-                       avroAsJsonSerialization: Option[Boolean] = None
-                      ) {
+                       avroAsJsonSerialization: Option[Boolean] = None,
+                       kafkaAddress: Option[String] = None) {
 
-  def schemaRegistryClientKafkaConfig = SchemaRegistryClientKafkaConfig(kafkaProperties.getOrElse(Map.empty), schemaRegistryCacheConfig, avroAsJsonSerialization)
+  // delete this method, make `kafkaProperties` required and use it instead, when `kafkaAddress` is removed
+  def definedKafkaProperties = kafkaAddress.map(bs => Map("bootstrap.servers" -> bs)).getOrElse(Map.empty) ++ kafkaProperties.getOrElse(Map.empty)
```

Comment:
maybe we can skip adding this method, and add address if needed only in KafkaUtils.withPropertiesFromConfig?

Reply:
but purpose of this method is also to get config as `Map[String,String]`, now both `kafkaAddress` and `kafkaProperties` are Options. 

If I was to delete this method, I would have to do in several places:  `kafkaProperties.getOrElse(Map.empty)` or change api in non-compatible way - that is change to: `kafkaProperties: Map[String,String]`

Reply:
Deleted, please see the last commit

Reply:
thx